olp-cpp-sdk-dataservice-write/test/unit/TestVersionedLayerClient.cpp

```diff
@@ -505,7 +518,13 @@ TEST_P(VersionedLayerClientOnlineTest, PublishToBatchMultiTest) {
       break;
     }
   }
-  ASSERT_EQ("succeeded", getBatchResponse.GetResult().GetDetails()->GetState());
+  // There are can be a case that GetBatch() is not "succeeded", but in
+  // "submitted" state even after 100 iterations,
+  // which actually means that there are might be a problem on the server side,
+  // (or just long delay). Thus, better to rewrite this test, or do not rely on
+  // the real server, but use mocked server.
+  // ASSERT_EQ("succeeded",
+  // getBatchResponse.GetResult().GetDetails()->GetState());
```

Comment:
In the first place, should the test even check for value set by a server?
It is OK to analyze if the field is present since SDK parses it from response JSON. But the values "submitted" or "succeeded" are not in the SDKs responsibility.

I propose to remove the checks for details' field completely from out (online) unit tests projects, for all layers affected.